api/v1beta1/common_types.go

```diff
@@ -46,6 +46,10 @@ const (
 	// to track the name of the MachineDeployment topology it represents.
 	ClusterTopologyMachineDeploymentLabelName = "topology.cluster.x-k8s.io/deployment-name"
 
+	// ClusterTopologyAdoptionAnnotation is the annotation used to allow adaption from an unmanaged cluster
+	// to a managed cluster with ClusterClass
+	ClusterTopologyAdoptionAnnotation = "topology.cluster.x-k8s.io/adoption"
+
```

Comment:
As per discussion in https://github.com/kubernetes-sigs/cluster-api/issues/6251 IMO the annotation name should explicitly call out the fact that there are no support guarantees at this stage.
Probably the comment should echo this as well. I will come out with some suggestion ASAP

Reply:
yes, the call-out was done below as the comments for ClusterTopologyAdoptionTypeManual. I will add to here as well.